Port IgnoreAuthenticationContextInResponse compatibility option to 2.x

explunit opened the pull request:
Fixes #1045

See also #780 for the original 1.x feature

AndersAbel commented on Sustainsys.Saml2/SAML2P/Saml2PSecurityTokenHandler.cs:
```diff
@@ -17,10 +17,15 @@ namespace Sustainsys.Saml2.Saml2P
-    {
-		public Saml2PSecurityTokenHandler()
+	{
+		public Saml2PSecurityTokenHandler(SPOptions spOptions)
```
This is a breaking change as the Saml2PSecurityTokenHandler is a public class.

I'd prefer if the old param-less ctor is kept side-by-side and thus the options has to be optional. Which means optional also for the Saml2PSerializer...

What do you think?

AndersAbel commented on Sustainsys.Saml2/SAML2P/Saml2PSecurityTokenHandler.cs:
```diff
@@ -17,10 +17,15 @@ namespace Sustainsys.Saml2.Saml2P
-			Serializer = new Saml2PSerializer();
+			if (spOptions == null)
+			{
+				throw new ArgumentNullException(nameof(spOptions));
```
Missing test coverage on this line - but let's figure out how to deal with the change of signature first.

explunit commented:
Fixed to allow the default constructor in Saml2PSecurityTokenHandler for backward compatibility. Since Saml2PSerializer is private I did not add a default constructor but I did fix the usage of spOptions to handle null (and added test coverage).
